open-craft/opencraft
# BB-2027: New API endpoints for frontend

This PR implements the following features:
- Adds new `deployments` endpoint, that allows retrieving status of deployment and cancelling latest deployments.
- Fixes Swagger generated documentation
- Toggles a flag that deactivates old appservers for self-deployed instances.

I went with a simple approach and just added the minimally required endpoints/functionality.

**Testing instructions:**
1. Checkout this branch.
2. Run Ocim.
3. Go to the swagger documentation (http://localhost:5000/api/swagger/)
4. Check the new `deployment` endpoints.
5. Create a new betatest instance.
9. Check the status endpoint.
6. Try to cancel the deployment of the first appserver, it should fail.
9. Check the status endpoint.
7. Create a mock `active` appserver.
8. Spawn a new appserver and try stopping the provisioning by making a `DELETE` request to the deployments endpoint.
9. Check the status endpoint.

**Reviewer:**

**Notes:**
We don't currently have any way of separating user-triggered deployments from OpenCraft deployments, so the user will be able to stop any provisioning for now. Any ideas of how we can implement this easily?
